File: tuan02/bai32.jl
function la_chinh_phuong(n::Int)
    # TODO
    if n < 0
        return false
    end
    
    left = Int128(0)
    right = Int128(10^9)
    
    while left <= right
        mid = (left + right) ÷ 2
        sq = mid * mid
        
        if sq == n
            return true
        elseif sq < n
            left = mid + 1
        else
            right = mid - 1
        end
    end
    
    return false
end

# Đoạn code chạy thử/nhập xuất:
n = parse(Int, readline())
println(la_chinh_phuong(n) ? "YES" : "NO")

File: tuan02/bai32.py
def la_chinh_phuong(n: int) -> bool:
    # TODO
    if n < 0:
        return False
        
    left, right = 0, 10**9
    while left <= right:
        mid = (left + right) // 2
        sq = mid * mid
        
        if sq == n:
            return True
        elif sq < n:
            left = mid + 1
        else:
            right = mid - 1
            
    return False

if __name__ == "__main__":
    n = int(input())
    print("YES" if la_chinh_phuong(n) else "NO")